### ISSUE TITLE
MariaDB: After 1.3.x upgrade: Unknown column "lft" on Rackgroup and Region

### ISSUE THREAD

**Issue by herr-mhet** [nautobot/nautobot#1724]
### Environment
* Python version: 3.9.2
* Nautobot version: 1.3.3

### Steps to Reproduce
1. Upgrade to 1.3.x
2. Visit either Rackgroups or Regions
3. See error message "(1054, "Unknown column 'U1.lft' in 'where clause'")"

### Expected Behavior
I can visit the Rackgroups and Regions pages.

### Observed Behavior
The aforementioned error message.

### Additional Info
This is a consistent error that I had trying to upgrade to 1.3.0 and now with 1.3.3. I did the "post_upgrade" and the "migrate" routines, so that is ruled out. I've migrated from netbox wayback, so this might affect the behaviour. Knowing this, I tried to manually remove the "lft" and "rght" columns from the dcim_region table, which did not alleviate this problem.

**Comment by bryanculver** [comment]
Thanks for the report @herr-mhet!

What version of Nautobot are you upgrading from?

**Comment by herr-mhet** [comment]
Thanks for getting back to me.

It's the version 1.2.11

**Comment by bryanculver** [comment]
This could be a problem stemming from the NetBox migration. Can you provide the full stack trace of the error message, potentially also with a screenshot? This is not immediately reproducible.

**Comment by herr-mhet** [comment]
Hi, sorry for not getting back to you sooner. 

Here is the stacktrace:

    May 18 11:02:47 nautobot nautobot-server[41076]: Internal Server Error: /dcim/rack-groups/
    May 18 11:02:47 nautobot nautobot-server[41076]: Traceback (most recent call last):
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    May 18 11:02:47 nautobot nautobot-server[41076]:     return self.cursor.execute(sql, params)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/db/backends/mysql/base.py", line 73, in execute
    May 18 11:02:47 nautobot nautobot-server[41076]:     return self.cursor.execute(query, args)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/MySQLdb/cursors.py", line 206, in execute
    May 18 11:02:47 nautobot nautobot-server[41076]:     res = self._query(query)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/MySQLdb/cursors.py", line 319, in _query
    May 18 11:02:47 nautobot nautobot-server[41076]:     db.query(q)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/MySQLdb/connections.py", line 254, in query
    May 18 11:02:47 nautobot nautobot-server[41076]:     _mysql.connection.query(self, query)
    May 18 11:02:47 nautobot nautobot-server[41076]: MySQLdb._exceptions.OperationalError: (1054, "Unknown column 'dcim_rackgroup.lft' in 'where clause'")
    May 18 11:02:47 nautobot nautobot-server[41076]: The above exception was the direct cause of the following exception:
    May 18 11:02:47 nautobot nautobot-server[41076]: Traceback (most recent call last):
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    May 18 11:02:47 nautobot nautobot-server[41076]:     response = get_response(request)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    May 18 11:02:47 nautobot nautobot-server[41076]:     response = wrapped_callback(request, *callback_args, **callback_kwargs)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/views/generic/base.py", line 70, in view
    May 18 11:02:47 nautobot nautobot-server[41076]:     return self.dispatch(request, *args, **kwargs)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/nautobot/utilities/views.py", line 95, in dispatch
    May 18 11:02:47 nautobot nautobot-server[41076]:     return super().dispatch(request, *args, **kwargs)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/views/generic/base.py", line 98, in dispatch
    May 18 11:02:47 nautobot nautobot-server[41076]:     return handler(request, *args, **kwargs)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/nautobot/core/views/generic.py", line 248, in get
    May 18 11:02:47 nautobot nautobot-server[41076]:     RequestConfig(request, paginate).configure(table)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django_tables2/config.py", line 61, in configure
    May 18 11:02:47 nautobot nautobot-server[41076]:     table.paginate(**kwargs)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django_tables2/tables.py", line 584, in paginate
    May 18 11:02:47 nautobot nautobot-server[41076]:     self.page = self.paginator.page(page)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/core/paginator.py", line 76, in page
    May 18 11:02:47 nautobot nautobot-server[41076]:     number = self.validate_number(number)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/core/paginator.py", line 54, in validate_number
    May 18 11:02:47 nautobot nautobot-server[41076]:     if number > self.num_pages:
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    May 18 11:02:47 nautobot nautobot-server[41076]:     res = instance.__dict__[self.name] = self.func(instance)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/core/paginator.py", line 103, in num_pages
    May 18 11:02:47 nautobot nautobot-server[41076]:     if self.count == 0 and not self.allow_empty_first_page:
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    May 18 11:02:47 nautobot nautobot-server[41076]:     res = instance.__dict__[self.name] = self.func(instance)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/core/paginator.py", line 98, in count
    May 18 11:02:47 nautobot nautobot-server[41076]:     return len(self.object_list)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django_tables2/rows.py", line 325, in __len__
    May 18 11:02:47 nautobot nautobot-server[41076]:     length = len(self.data)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django_tables2/data.py", line 153, in __len__
    May 18 11:02:47 nautobot nautobot-server[41076]:     self._length = self.data.count()
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/cacheops/query.py", line 294, in count
    May 18 11:02:47 nautobot nautobot-server[41076]:     return self._no_monkey.count(self)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/db/models/query.py", line 412, in count
    May 18 11:02:47 nautobot nautobot-server[41076]:     return self.query.get_count(using=self.db)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/db/models/sql/query.py", line 528, in get_count
    May 18 11:02:47 nautobot nautobot-server[41076]:     number = obj.get_aggregation(using, ['__count'])['__count']
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/db/models/sql/query.py", line 513, in get_aggregation
    May 18 11:02:47 nautobot nautobot-server[41076]:     result = compiler.execute_sql(SINGLE)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
    May 18 11:02:47 nautobot nautobot-server[41076]:     cursor.execute(sql, params)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/cacheops/transaction.py", line 97, in execute
    May 18 11:02:47 nautobot nautobot-server[41076]:     result = self._no_monkey.execute(self, sql, params)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/db/backends/utils.py", line 66, in execute
    May 18 11:02:47 nautobot nautobot-server[41076]:     return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    May 18 11:02:47 nautobot nautobot-server[41076]:     return executor(sql, params, many, context)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    May 18 11:02:47 nautobot nautobot-server[41076]:     return self.cursor.execute(sql, params)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/db/utils.py", line 90, in __exit__
    May 18 11:02:47 nautobot nautobot-server[41076]:     raise dj_exc_value.with_traceback(traceback) from exc_value
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    May 18 11:02:47 nautobot nautobot-server[41076]:     return self.cursor.execute(sql, params)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/django/db/backends/mysql/base.py", line 73, in execute
    May 18 11:02:47 nautobot nautobot-server[41076]:     return self.cursor.execute(query, args)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/MySQLdb/cursors.py", line 206, in execute
    May 18 11:02:47 nautobot nautobot-server[41076]:     res = self._query(query)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/MySQLdb/cursors.py", line 319, in _query
    May 18 11:02:47 nautobot nautobot-server[41076]:     db.query(q)
    May 18 11:02:47 nautobot nautobot-server[41076]:   File "/opt/nautobot/lib/python3.9/site-packages/MySQLdb/connections.py", line 254, in query
    May 18 11:02:47 nautobot nautobot-server[41076]:     _mysql.connection.query(self, query)
    May 18 11:02:47 nautobot nautobot-server[41076]: django.db.utils.OperationalError: (1054, "Unknown column 'dcim_rackgroup.lft' in 'where clause'")

The relevant database tables are empty:

    MariaDB [nautobot]> select * from dcim_rackgroup limit 1;
    Empty set (0.000 sec)
    MariaDB [nautobot]> select * from tenancy_tenantgroup limit 1;
    Empty set (0.000 sec)

This is how the tables are structured:

    MariaDB [nautobot]> describe dcim_rackgroup;
    +--------------------+------------------+------+-----+---------+-------+
    | Field              | Type             | Null | Key | Default | Extra |
    +--------------------+------------------+------+-----+---------+-------+
    | id                 | char(32)         | NO   | PRI | NULL    |       |
    | created            | date             | YES  |     | NULL    |       |
    | last_updated       | datetime(6)      | YES  |     | NULL    |       |
    | _custom_field_data | longtext         | NO   |     | NULL    |       |
    | name               | varchar(100)     | NO   | MUL | NULL    |       |
    | slug               | varchar(100)     | NO   | MUL | NULL    |       |
    | description        | varchar(200)     | NO   |     | NULL    |       |
    | lft                | int(10) unsigned | NO   |     | NULL    |       |
    | rght               | int(10) unsigned | NO   |     | NULL    |       |
    | tree_id            | int(10) unsigned | NO   | MUL | NULL    |       |
    | level              | int(10) unsigned | NO   |     | NULL    |       |
    | parent_id          | char(32)         | YES  | MUL | NULL    |       |
    | site_id            | char(32)         | NO   | MUL | NULL    |       |
    +--------------------+------------------+------+-----+---------+-------+
    MariaDB [nautobot]> describe tenancy_tenantgroup;
    +--------------------+------------------+------+-----+---------+-------+
    | Field              | Type             | Null | Key | Default | Extra |
    +--------------------+------------------+------+-----+---------+-------+
    | id                 | char(32)         | NO   | PRI | NULL    |       |
    | created            | date             | YES  |     | NULL    |       |
    | last_updated       | datetime(6)      | YES  |     | NULL    |       |
    | _custom_field_data | longtext         | NO   |     | NULL    |       |
    | name               | varchar(100)     | NO   | UNI | NULL    |       |
    | slug               | varchar(100)     | NO   | UNI | NULL    |       |
    | description        | varchar(200)     | NO   |     | NULL    |       |
    | lft                | int(10) unsigned | NO   |     | NULL    |       |
    | rght               | int(10) unsigned | NO   |     | NULL    |       |
    | tree_id            | int(10) unsigned | NO   | MUL | NULL    |       |
    | level              | int(10) unsigned | NO   |     | NULL    |       |
    | parent_id          | char(32)         | YES  | MUL | NULL    |       |
    +--------------------+------------------+------+-----+---------+-------+

**Comment by glennmatthews** [comment]
Still chasing this down, but it looks to be specific to MariaDB - we can reproduce this issue locally with it, while no issue is seen with MySQL or Postgres.

**Comment by bryanculver** [comment]
Running our unit tests on MariaDB shows several errors:

```
➜ invoke unittest
Running docker-compose command "ps --services --filter status=running"
Running docker-compose command "exec nautobot mkdocs build --no-directory-urls --strict"
INFO     -  Cleaning site directory
INFO     -  Building documentation to directory: /source/nautobot/project-static/docs
Running docker-compose command "ps --services --filter status=running"
Running docker-compose command "exec nautobot coverage run --module nautobot.core.cli --config=nautobot/core/tests/nautobot_config.py test nautobot --buffer"
Creating test database for alias 'default'...

System check identified some issues:

WARNINGS:
<Settings "nautobot_config">: (nautobot.core.W005) STORAGE_CONFIG has been set but STORAGE_BACKEND is not defined. STORAGE_CONFIG will be ignored.
extras.Job.slug: (mysql.W003) MariaDB may not allow unique CharFields to have a max_length > 255.
	HINT: See: https://docs.djangoproject.com/en/3.2/ref/databases/#mysql-character-fields
extras.Job.slug: (mysql.W003) MariaDB may not allow unique CharFields to have a max_length > 255.
	HINT: See: https://docs.djangoproject.com/en/3.2/ref/databases/#mysql-character-fields

System check identified 3 issues (0 silenced).
...................s.........................................s....................s.........
...........s................................................................................
............................................................................................
........................................................................s...................
....................................s....................s.....................s............
........s.....................s....................s....................s...................
............................s....................s....................s....................s
....................s.....................s....................s....................s.......
..................................s....................s.....................s..............
......s....................s...FFFF.F.FF.FFF.....sF...................s....................s
..............................................s....................s..FFFF.F.FF.FFF.....sF..
.............................................s..............................................
............................................................................................
............................................................................................
............................................................................................
............................................................................................
............................................................................................
............................................................................................
............................................................................................
............................................................................................
............................................................................................
............................................................................................
........EE...................EEE............................................................
.................................................................................EE.........
..........EEE.....................E.........................................................
....s....................s.......................s............................s....s........
.......s....s................s....................s.........................s....s..........
......s................s...........................s........................................
...............s.........................s...............s....................s.............
s..................s....................s....................s....................s.........
............s.............s..................s..............................................
............................................................................................
............................................................................................
........................................s...................................................
............................................................................................
............................................................................................
............................................................................................
............................................................................................
............................................................................................
....................................................................s.......................
................................................s....................s....................s.
....................s....................s..........................................s.......
............................................................................................
......................................................s.....................................
............................................................................................
............................................................................................
............................................................................................
.......FFFF.F.FF.FFF.....sF...................s............EE...................EEE.........
...........................s...............s....................s...........................
s......s...............s....................................................................
............................................................................................
....................................................................s....................s..
..................s....................s....................................................
............................................................................................
...............................................................................EF
Stdout:

F
Stdout:

F
Stdout:

F
Stdout:

F
Stdout:

F
Stdout:

F
Stdout:

F
Stdout:

F
Stdout:

F
Stdout:

E.F
Stdout:

F
Stdout:

F
Stdout:

F
Stdout:

F
Stdout:

F
Stdout:

F
Stdout:

.F
Stdout:

F
Stdout:

F
Stdout:

F
Stdout:

.F
Stdout:

....
```

**Comment by jathanism** [comment]
Although we cannot officially support MariaDB at this time, we did do a little research to address this issue for you.

I confirmed that if you explicitly `pip install django-mptt==0.11.0` for your Nautobot installation, that this issue is alleviated.

Since there are no other dependencies or conflicts with `django-mptt`, this should be fine for now in your deployment. Also, we do plan to get rid of this library in our v2.0.0 release, so this issue will also no longer be a concern.

I'll be closing this issue. Thanks for reporting it!
